src/foam/dao/JDAO.java

```diff
@@ -7,24 +7,117 @@
 package foam.dao;
 
 import foam.core.ClassInfo;
-import foam.lib.json.Outputter;
-import foam.lib.json.OutputterMode;
+import foam.core.FObject;
+import foam.core.X;
+import foam.mlang.order.Comparator;
+import foam.mlang.predicate.Predicate;
+import foam.nanos.auth.User;
+import foam.util.SafetyUtil;
+
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.TimeZone;
 
 public class JDAO
-  extends AbstractJDAO
+  extends ProxyDAO
 {
-  protected Outputter outputter_ = new Outputter(OutputterMode.STORAGE);
+  protected static ThreadLocal<StringBuilder> sb = new ThreadLocal<StringBuilder>() {
+    @Override
+    protected StringBuilder initialValue() {
+      return new StringBuilder();
+    }
+
+    @Override
+    public StringBuilder get() {
+      StringBuilder b = super.get();
+      b.setLength(0);
+      return b;
+    }
+  };
+
+  protected static final ThreadLocal<SimpleDateFormat> sdf = new ThreadLocal<SimpleDateFormat>() {
+    @Override
+    protected SimpleDateFormat initialValue() {
+      SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'");
+      sdf.setTimeZone(TimeZone.getTimeZone("UTC"));
+      return sdf;
+    }
+  };
 
-  public JDAO(foam.core.X x, ClassInfo classInfo, String filename) {
+  protected FileJournal journal_;
+
+  public JDAO(X x, ClassInfo classInfo, String filename) {
     this(x, new MapDAO(classInfo), filename);
   }
 
-  public JDAO(foam.core.X x, DAO delegate, String filename) {
-    super(x, delegate, filename);
+  public JDAO(X x, DAO delegate, String filename) {
```

Comment:
The constructor for this should just take a journal. These convenience constructors should be removed and done using the facade pattern.